src/artifact/tar/platform/libarchive/wrapper.cpp

```diff
@@ -38,15 +38,15 @@ using ExpectedSize = expected::ExpectedSize;
 // the size of the data blocks returned.
 //
 // - On EOF return 0.
-// - On error return -1.
+// - On error, set archive_set_error and return ARCHIVE_FATAL.
 ssize_t reader_callback(archive *archive, void *in_reader_container, const void **buff) {
 	ReaderContainer *p_reader_container = static_cast<ReaderContainer *>(in_reader_container);
 
 	auto ret = p_reader_container->reader_.Read(
 		p_reader_container->buff_.begin(), p_reader_container->buff_.end());
 	if (!ret) {
-		log::Error("Failed to read from the archive stream: Error: " + ret.error().message);
-		return -1;
+		archive_set_error(archive, ret.error().code.value(), "%s", ret.error().message.c_str());
+		return ARCHIVE_FATAL;
```

Comment:
The `read` callback should actually return `-1` in case of error. It's the **`open`** callback that should return `ARCHIVE_FATAL`, but we don't use `open` callback.

Reply:
Thank you.